okex-price/src/pricefeed/feeder.rs

```diff
@@ -0,0 +1,107 @@
+use super::{error::PriceFeedError, PriceFeedConfig};
+use futures::{
+    stream::{SplitSink, SplitStream},
+    SinkExt, Stream, StreamExt,
+};
+use serde::{Deserialize, Serialize};
+use shared::{
+    payload,
+    pubsub::{Envelope, MessagePayload},
+};
+use std::pin::Pin;
+use tokio::net::TcpStream;
+use tokio_tungstenite::{connect_async, tungstenite::Message, MaybeTlsStream, WebSocketStream};
+use url::Url;
+
+#[derive(Clone, Serialize, Deserialize, Debug)]
+#[serde(rename_all = "camelCase")]
+struct ChannelData {
+    channel: String,
+    inst_id: String,
+}
+
+#[derive(serde::Serialize)]
+struct ChannelSubscription {
+    op: String,
+    args: Vec<ChannelData>,
+}
+
+impl ChannelSubscription {
+    fn new() -> Self {
+        Self {
+            op: "subscribe".to_string(),
+            args: vec![ChannelData {
+                channel: "tickers".to_string(),
+                inst_id: "BTC-USD-SWAP".to_string(),
+            }],
+        }
+    }
+}
+
+#[derive(Clone, Serialize, Deserialize, Debug)]
+#[serde(rename_all = "camelCase")]
+struct TickersChannelData {
+    inst_type: String,
+    inst_id: String,
+    last: String,
+    last_sz: String,
+    ask_px: String,
+    ask_sz: String,
+    bid_px: String,
+    bid_sz: String,
+    open24h: String,
+    high24h: String,
+    low24h: String,
+    vol_ccy24h: String,
+    vol24h: String,
+    sod_utc0: String,
+    sod_utc8: String,
+    ts: String,
+}
+
+#[derive(Clone, Serialize, Deserialize, Debug)]
+pub struct OkexPriceTick {
+    arg: ChannelData,
+    data: Vec<TickersChannelData>,
+}
+
+payload! { OkexPriceTick, "okex.price" }
+
+#[derive(Debug)]
+pub struct PriceFeed {
```

Comment:
I guess we don't need the PriceFeed struct as we cannot re-use the receiver.

I guess we can have a top level:
src/price-feed/mod.rs
```
pub fn subscribe_btc_usd_swap()
```
or something like that

Reply:
I have implemented the `subscribe_btc_usd_swap()` function as recommended.